### PR TITLE
Color Slack messages from jobs based on title then body contents

### PR DESCRIPTION
**Pull request by yoomlam**
### Description
Warning message from jobs are shown as red because the message body contains the word "error" ([example1](https://dsva.slack.com/archives/CN3FQR4A1/p1604588199012500), [example2](https://dsva.slack.com/archives/CN3FQR4A1/p1604398921004900)). This PR distinguishes the Slack title from the Slack message body and colors based on the Slack title first.

### Acceptance Criteria
- [ ] Code compiles correctly

### Testing Plan
Check #appeals-job-alerts to check message coloring.

### CODE REVIEW

File: app/jobs/set_appeal_age_aod_job.rb [hunk 1]
```diff
@@ -30,21 +30,21 @@ def perform
 
   def log_success(details)
     duration = time_ago_in_words(start_time)
-    msg = "#{self.class.name} completed after running for #{duration}.\n#{details}"
-    Rails.logger.info(msg)
+    msg_title = "[INFO] #{self.class.name} completed after running for #{duration}."
+    Rails.logger.info("#{msg_title}\n#{details}")
 
-    slack_service.send_notification("[INFO] #{msg}")
+    slack_service.send_notification(details, msg_title)
```
Comment (yoomlam) [inline app/jobs/set_appeal_age_aod_job.rb:36]: Split Slack title from message body.

File: app/jobs/stats_collector_job.rb [hunk 2]
```diff
@@ -96,7 +96,7 @@ def log_success
     msg = "#{self.class.name} completed after running for #{duration}."
     Rails.logger.info(msg)
 
-    slack_service.send_notification("[INFO] #{msg}") # may not need this
+    slack_service.send_notification("[INFO] #{msg}", self.class.to_s) # may not need this
```
Comment (yoomlam) [inline app/jobs/stats_collector_job.rb:99]: Existing jobs use the class as the msg title.

File: app/jobs/update_cached_appeals_attributes_job.rb [hunk 3]
```diff
@@ -121,9 +120,9 @@ def log_error(start_time, err)
 
     Raven.capture_exception(err)
 
-    slack_msg = "[ERROR] UpdateCachedAppealsAttributesJob failed after running for #{duration}. "\
-                "See Sentry event #{Raven.last_event_id}"
-    slack_service.send_notification(slack_msg) # do not leak PII
+    slack_msg = "See Sentry event #{Raven.last_event_id}"
+    slack_service.send_notification(slack_msg,
+                                    "[ERROR] UpdateCachedAppealsAttributesJob failed after running for #{duration}.")
```
Comment (yoomlam) [inline app/jobs/update_cached_appeals_attributes_job.rb:125]: Split Slack title from message body.

File: app/services/slack_service.rb [hunk 4]
```diff
@@ -36,9 +36,13 @@ def http_service
...
   def pick_color(title, msg)
-    if title =~ /error/i || msg =~ /error/i
+    if /error/i.match?(title)
       COLORS[:error]
-    elsif title =~ /warn/i || msg =~ /warn/i
+    elsif /warn/i.match?(title)
+      COLORS[:warn]
+    elsif /error/i.match?(msg)
+      COLORS[:error]
+    elsif /warn/i.match?(msg)
```
Comment (yoomlam) [inline app/services/slack_service.rb:45]: Crux of this PR: match against the title first, then check the msg body.

File: spec/services/slack_service_spec.rb [hunk 5]
```diff
@@ -38,6 +38,13 @@
       end
     end
 
+    context "message contains error but title contains warning" do
+      it "picks yellow color" do
+        slack_service.send_notification("there was an error", "Really just a warning")
+        expect(@http_params[:body]).to match(/"#ffff00"/)
+      end
+    end
+
```
Comment (yoomlam) [inline spec/services/slack_service_spec.rb:47]: New test for crux of PR.

**Review by rubaiyat22 — APPROVED** [review]
:shipit:

**Pull request merged**